dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,352 @@
+package com.dotcms.security.secret;
+
+import com.dotcms.enterprise.cluster.ClusterFactory;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.util.Encryptor;
+import com.liferay.util.FileUtil;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+import java.io.File;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStore.PasswordProtection;
+import java.security.KeyStore.SecretKeyEntry;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+
+/**
+ * https://stackoverflow.com/questions/6243446/how-to-store-a-simple-key-string-inside-java-keystore
+ * https://medium.com/@danojadias/aes-256bit-encryption-decryption-and-storing-in-the-database-using-java-2ada3f2a0b14
+ * https://neilmadden.blog/2017/11/17/java-keystores-the-gory-details/
+ *
+ */
+public class SecretsStoreKeyStoreImpl implements SecretsStore {
```

Comment:
I think this doc is not good enough

Reply:
+1

In addition we used to separated the Cache into another class instead of mixing in the service or factory

Reply:
I think also a Factory could be created from the code, it seems cache, persistence and biz logic is all mix